homeassistant/components/sensor/hive.py

```diff
@@ -0,0 +1,486 @@
+                        temperature_target})
+            s_a.update({"Temperature Difference":
+                        temperature_difference})
+
+        return s_a
+
+    def get_heating_state_sa(self):
+        """Public get current heating state, state attributes."""
+        s_a = {}
+
+        snan = self.session.heating.get_schedule_now_next_later(self.node_id)
+        if snan is not None:
+            if 'now' in snan:
+                if ('value' in snan["now"] and
+                        'start' in snan["now"] and
+                        'Start_DateTime' in snan["now"] and
+                        'End_DateTime' in snan["now"] and
+                        'target' in snan["now"]["value"]):
+                    now_target = str(snan["now"]["value"]["target"]) + " °C"
+                    nstrt = snan["now"]["Start_DateTime"].strftime("%H:%M")
+                    now_end = snan["now"]["End_DateTime"].strftime("%H:%M")
+
+                    sa_string = (now_target
+                                 + " : "
+                                 + nstrt
+                                 + " - "
+                                 + now_end)
+                    s_a.update({"Now": sa_string})
+
+            if 'next' in snan:
+                if ('value' in snan["next"] and
+                        'start' in snan["next"] and
+                        'Start_DateTime' in snan["next"] and
+                        'End_DateTime' in snan["next"] and
+                        'target' in snan["next"]["value"]):
+                    next_target = str(snan["next"]["value"]["target"]) + " °C"
+                    nxtstrt = snan["next"]["Start_DateTime"].strftime("%H:%M")
+                    next_end = snan["next"]["End_DateTime"].strftime("%H:%M")
+
+                    sa_string = (next_target
+                                 + " : "
+                                 + nxtstrt
+                                 + " - "
+                                 + next_end)
+                    s_a.update({"Next": sa_string})
+
+            if 'later' in snan:
+                if ('value' in snan["later"] and
+                        'start' in snan["later"] and
+                        'Start_DateTime' in snan["later"] and
+                        'End_DateTime' in snan["later"] and
+                        'target' in snan["later"]["value"]):
+                    ltarg = str(snan["later"]["value"]["target"]) + " °C"
+                    lstrt = snan["later"]["Start_DateTime"].strftime("%H:%M")
+                    lend = snan["later"]["End_DateTime"].strftime("%H:%M")
+
+                    sa_string = (ltarg
+                                 + " : "
+                                 + lstrt
+                                 + " - "
+                                 + lend)
+                    s_a.update({"Later": sa_string})
+        else:
+            s_a.update({"Schedule not active": ""})
+
+        return s_a
+
+    def get_hotwater_state_sa(self):

```

Comment:
Same as above.

Reply:
The data gathering is already in the library. the code here just converts to HA specific formatting